Core/Commands/CommandProcessor.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using ArgsParsing;
+using Microsoft.Extensions.Logging;
+
+namespace Core.Commands
+{
+    /// <summary>
+    /// The command processor can be configured using <see cref="CommandInfo"/> instances to have commands,
+    /// which then get executed using the <see cref="CommandProcessor.Process"/> method.
+    /// </summary>
+    public class CommandProcessor
```

Comment:
it might be worth having a system where you can have a "cooldown" for commands

Reply:
any specific requirements? like per-user cooldowns? or cross-command cooldowns? or just a fixed global cooldown per command?

Reply:
optional cooldown, per user might be good idea as an extra to a global